src/Connectors/src/Connector/CosmosDb/CosmosDbWebApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using System.Data.Common;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Steeltoe.Common;
+using Steeltoe.Common.HealthChecks;
+
+namespace Steeltoe.Connector.CosmosDb;
+
+public delegate object CreateCosmosClient(CosmosDbOptions options, string serviceBindingName);
+
+public static class CosmosDbWebApplicationBuilderExtensions
+{
+    private static readonly Type ConnectionType = CosmosDbTypeLocator.CosmosClient;
+
+    public static WebApplicationBuilder AddCosmosDb(this WebApplicationBuilder builder)
+    {
+        // CosmosClient has no constructor with only connection string parameter, so pass null for CosmosClientOptions parameter.
```

Comment:
Should we expose a way to configure `CosmosClientOptions` instead of passing null, or you feel that use case is sufficiently handled with the `AddCosmosDb` option below this one?

Reply:
Yes, that's why the second overload is public. Snapshot from the updated sample:
```c#
builder.AddCosmosDb((options, serviceBindingName) =>
{
    var cosmosClientBuilder = new CosmosClientBuilder(options.ConnectionString);
    cosmosClientBuilder.WithApplicationName("cosmosdb-connector");
    return cosmosClientBuilder.Build();
});
```